### DISCUSSION TITLE
"Please start a new Shell for changes to take effect"

### DISCUSSION THREAD

**Discussion by aghasemi** [jbangdev/jbang#1697] [Q&A]
Hi,

This is the last(?) message when installing JBang. But what if we are installing JBang in a Dockerfile and later in the same Dockerfile we want to run the `jbang` command, e.g. to set `JAVA_HOME`? Is there a solution here?

In my case, `eval $(jbang jdk java-env)` failed the docker build because of this situation.

**Comment by maxandersen** [comment]
source ~/.bashrc should do the trick.

**Comment by maxandersen** [comment]
source ~/.bashrc should do the trick. 

Minimally just add ~/.jbang/bin to your PATH.

**Comment by aghasemi** [comment]
Thanks

**Comment by aghasemi** [comment]
Hi. For future reference, merely running `source` does not work by default. One has to add `SHELL ["/bin/bash", "-c"]` to the beginning of the Dockerfile first.
